httpcore5/src/main/java/org/apache/hc/core5/reactor/ssl/SSLIOSession.java

```diff
@@ -588,14 +588,14 @@ private void decryptData(final IOSession protocolSession) throws IOException {
                         if (sslEngine.isInboundDone()) {
                             endOfStream = true;
                         }
-                        if (inPlainBuf.hasRemaining()) {
```

Comment:
@JasonMathison Oh, damn. This check should have been `inPlainBuf.position() > 0`. The idea is to avoid doing the buffer flip and a useless input event with an empty buffer. Could you please test code with this change and let me know if everything still works as expected?